Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCALayerRemote.mm

```diff
@@ -189,11 +189,28 @@
     clone.updateCustomAppearance(customAppearance());
 }
 
+void PlatformCALayerRemote::recursiveMarkWillBeDisplayed()
+{
+    if (m_properties.backingStoreOrProperties.store && m_properties.backingStoreAttached)
+        m_properties.backingStoreOrProperties.store->layerWillBeDisplayed();
+
+    for (size_t i = 0; i < m_children.size(); ++i) {
+        PlatformCALayerRemote& child = downcast<PlatformCALayerRemote>(*m_children[i]);
+        ASSERT(child.superlayer() == this);
+        child.recursiveMarkWillBeDisplayed();
+    }
+}
+
 void PlatformCALayerRemote::recursiveBuildTransaction(RemoteLayerTreeContext& context, RemoteLayerTreeTransaction& transaction)
 {
     ASSERT(!m_properties.backingStoreOrProperties.store || owner());
     RELEASE_ASSERT_WITH_SECURITY_IMPLICATION(&context == m_context);
-    
+
+    if (owner() && owner()->platformCALayerRenderingSuppressed()) {
+        recursiveMarkWillBeDisplayed();
```

Comment:
I'm confused by this. If rendering is suppressed on the PlatformCALayer (and that's just true for the root layer, right?) why are we marking the layer as "will be displayed"?